src/main/java/org/jabref/gui/citationkeypattern/GenerateCitationKeyAction.java

```diff
@@ -131,10 +141,10 @@ protected Void call() {
             public BackgroundTask<Void> onSuccess(Consumer<Void> onSuccess) {
                 // register the undo event only if new citation keys were generated
                 if (compound.hasEdits()) {
-                    frame.getUndoManager().addEdit(compound);
+                    undoManager.addEdit(compound);
                 }
 
-                frame.getCurrentLibraryTab().markBaseChanged();
+                tabSupplier.get().markBaseChanged();
```

Comment:
It looks like most of the remaining usages of the LibraryTab are now related to the "changed" status. Is it easy to push this down to the database context (as a follow-up of course)?

Reply:
We'll see. My plan is to somehow really fix the undo redo stuff and for that I want to utilize the commands we're using and maybe putting them eventually on a reactive stream. But there is still some road to walk.
Upmost next is some refactoring of libraryTab and jabrefframe